### ISSUE TITLE
Dynamic Segments with middleware stop working after 13.3.0

### ISSUE THREAD

**Issue by nicosh** [vercel/next.js#48568]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 11 Pro
    Binaries:
      Node: 18.15.0
      npm: 9.5.0
      Yarn: N/A
      pnpm: 7.29.3
    Relevant packages:
      next: 13.3.0
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Middleware / Edge (API routes, runtime), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue

https://github.com/nicosh/middleware-redirect-issue-repro

### To Reproduce

- Make sure to be on the main branch (next 13.2.4)
- npm install
- npm run dev
- Visit http://localhost:3000
- You should be able to see the page
- Change branch (checkout on _bug_ branch)
- npm install ( it will install next 13.3.0)
- npm run dev
- Visit http://localhost:3000
- It should return 404 page

The project structure is inspired by [platforms starter kit](https://github.com/vercel/platforms/tree/app-dir), routing and redirects with middleware works on next 13.2.4 but it breaks on 13.3.0 ( and all other canaries)

### Describe the Bug

visit http://localhost:3000 returns 404

### Expected Behavior

I'm expecting to see the page

### Which browser are you using? (if relevant)

chrome

**Comment by balazsorban44** [comment]
This is expected, please see https://github.com/vercel/next.js/pull/47439 for an explanation.

TLDR; you can rename the `_sites` directory to `%5Fsites`, which is the URL encoded version.
